cocotb/binary.py

```diff
@@ -96,22 +97,37 @@ class BinaryValue(object):
     _resolve_to_error = "xXzZuUwW"  # Resolve to a ValueError() since these usually mean something is wrong
     _permitted_chars  = _resolve_to_0 +_resolve_to_1 + _resolve_to_error + "01"  # noqa
 
-    def __init__(self, value=None, bits=None, bigEndian=True,
-                 binaryRepresentation=BinaryRepresentation.UNSIGNED):
+    def __init__(self, value=None, n_bits=None, bigEndian=True,
+                 binaryRepresentation=BinaryRepresentation.UNSIGNED,
+                 bits=None):
         """
         Kwagrs:
-            Value (string or int or long): value to assign to the bus
+            value (string or int or long): value to assign to the bus
 
-            bits (int): Number of bits to use for the underlying binary
-                        representation
+            n_bits (int): Number of bits to use for the underlying binary
+                          representation
 
             bigEndian (bool): Interpret the binary as big-endian when
                                 converting to/from a string buffer.
+
+            bits (int): Deprecated: Compatibility wrapper for n_bits
         """
         self._str = ""
-        self._bits = bits
         self.big_endian = bigEndian
         self.binaryRepresentation = binaryRepresentation
+
+        # bits is the deprecated name for n_bits, allow its use for
+        # backward-compat reasons.
+        if (bits is not None and n_bits is not None):
+            raise ValueError("You cannot use n_bits and bits at the same time.")
```

Comment:
Typically passing the wrong set of kwargs is considered a `TypeError`. For instance, try `def x(arg): pass`, `x(1, arg=1)`